src/store/modules/util/actions.ts

```diff
@@ -57,6 +58,79 @@ const actions: ActionTree<UtilState, RootState> = {
   async clearAppliedFilters({ commit }) {
     commit(types.UTIL_APPLIED_FILTERS_CLEARED)
   },
+
+  async fetchFacilities({ commit, dispatch, state }, payload) {
+    const params = {
+      parentFacilityTypeId: 'VIRTUAL_FACILITY',
+      parentFacilityTypeId_not: 'Y',
+      facilityTypeId: 'VIRTUAL_FACILITY',
+      facilityTypeId_not: 'Y',
+      productStoreId: store.state.user.currentEComStore.productStoreId,
+      pageSize: payload.pageSize,
+      pageIndex: payload.pageIndex
+    }
+
+    const facilities = state.facilities.list ? JSON.parse(JSON.stringify(state.facilities.list)) : [];
+    let isScrollable = true, facilityList = [];
+
+    try {
+      const resp = await UtilService.fetchFacilities(params)
+
+      if(!hasError(resp)) {
+        if(payload.pageIndex && payload.pageIndex > 0) {
+          facilityList = facilities.concat(resp.data)
+        } else {
+          facilityList = resp.data
+        }
+
+        if(payload.isOrderCountRequired) {
+          const facilityIds = resp.data.map((facility: any) => facility.facilityId)
+          const facilityCounts = await dispatch("fetchFacilitiesOrderCount", { facilityIds })
+
+          facilityList.map((facility: any) => {
+            if(facilityCounts[facility.facilityId]) facility.orderCount = facilityCounts[facility.facilityId]
+            else facility.orderCount = 0;
```

Comment:
This check will make the orderCount for the previous facilities to 0 when user scrolls the page.